packages/shared/src/features/profile/components/ProfileUploadBanner.tsx

```diff
@@ -0,0 +1,171 @@
+import type { ReactElement } from 'react';
+import React from 'react';
+import classNames from 'classnames';
+import type { MutationStatus } from '@tanstack/react-query';
+import { useRouter } from 'next/router';
+import {
+  Typography,
+  TypographyColor,
+  TypographyType,
+} from '../../../components/typography/Typography';
+import { DragDrop } from '../../../components/fields/DragDrop';
+import {
+  uploadCvBannerSuccessLaptop,
+  uploadCvBannerSuccessMobile,
+  uploadCvBannerSuccessTablet,
+  uploadCvBgLaptop,
+  uploadCvBgMobile,
+  uploadCvBgTablet,
+} from '../../../lib/image';
+import {
+  Button,
+  ButtonSize,
+  ButtonVariant,
+} from '../../../components/buttons/Button';
+import { MiniCloseIcon } from '../../../components/icons';
+import { useToastNotification, useViewSize, ViewSize } from '../../../hooks';
+import { cvUploadBannerBg } from '../../../styles/custom';
+import { FeelingLazy } from './FeelingLazy';
+import { webappUrl } from '../../../lib/constants';
+import { fileValidation } from '../hooks/useUploadCv';
+
+const defaultBanner = {
+  title: 'Your next job should apply to you',
+  description:
+    'Upload your CV so we quietly match you with roles you might actually want. Nothing is shared without your ok.',
+  cover: {
+    laptop: uploadCvBgLaptop,
+    tablet: uploadCvBgTablet,
+    base: uploadCvBgMobile,
+  },
+  successCover: {
+    laptop: uploadCvBannerSuccessLaptop,
+    tablet: uploadCvBannerSuccessTablet,
+    base: uploadCvBannerSuccessMobile,
+  },
+};
+
+interface Cover {
+  laptop: string;
+  tablet: string;
+  base: string;
+}
+
+interface ProfileUploadBannerProps {
+  banner?: {
+    title: string;
+    description: string;
+    cover?: Cover;
+    successCover?: Cover;
+  };
+  className?: Partial<{
+    container: string;
+    image: string;
+  }>;
+  onClose: () => void;
+  onUpload: (file: File) => Promise<void>;
+  status: MutationStatus;
+}
+
+export function ProfileUploadBanner({
+  className,
+  banner = defaultBanner,
+  onClose,
+  onUpload,
+  status,
+}: ProfileUploadBannerProps): ReactElement {
+  const justUploaded = status === 'success';
+  const isLaptop = useViewSize(ViewSize.Laptop);
+  const isTablet = useViewSize(ViewSize.Tablet);
+  const { displayToast } = useToastNotification();
+
+  const getImage = () => {
+    const cover = justUploaded
+      ? banner?.successCover || banner?.cover
+      : banner?.cover;
+
+    if (isLaptop) {
+      return cover?.laptop || uploadCvBgTablet;
+    }
+
+    if (isTablet) {
+      return cover?.tablet || uploadCvBgTablet;
+    }
+
+    return cover?.base || uploadCvBgMobile;
+  };
```

Comment:
Not related to this PR but maybe we should add a `useValueByViewSize(options: { [ViewSize]: unknow }, default)` that basically do: 

```ts

for(let viewport of options){
  if({is this viewport}) return options[viewport];
}

return default;

```